niri-ipc/src/lib.rs

```diff
@@ -1174,7 +1176,11 @@ impl FromStr for LayoutSwitchTarget {
         match s {
             "next" => Ok(Self::Next),
             "prev" => Ok(Self::Prev),
-            _ => Err(r#"invalid layout action, can be "next" or "prev""#),
+            other =>
+                match other.parse::<u32>() {
```

Comment:
```suggestion
                match other.parse() {
```
I think this should be able to infer itself?

Reply:
Done